.github/workflows/changelog-check.yaml

```diff
@@ -1,48 +1,59 @@
-name: Check CHANGELOG.md updated
-"on":
+name: Check for changelog entry file
+on:
   pull_request:
     types: [opened, synchronize, reopened, edited]
     # It's important to check that the changelog is updated with bug fixes that
     # we backport to the release branches, so these branches are included as
     # well.
-    branches: [main, '[0-9]+.[0-9]+.x']
+    branches: ["main", "[0-9]+.[0-9]+.x"]
 jobs:
-  # Check that the CHANGELOG is updated by the pull request. This can be
-  # disabled by adding a trailer line of the following form to the
-  # pull request message:
+  # Check if the PR creates a sperate file with changelog entry in the
+  # ".unreleased/features"  or ".unreleased/features"  folder
   #
-  #    Disable-Check: force-changelog-changed
+  # This check can be disabled by adding the following line in the PR text
   #
-  # The check is case-insensitive and ignores other contents on the
-  # line as well, so it is possible to add several different checks if
-  # that is necessary.
+  # Disable-check: force-changelog-file
   #
-  # It is assumed that the trailer is following RFC2822 conventions,
-  # but this is currently not enforced.
-  check_changelog_update:
-    name: Check for CHANGELOG.md changes
+  # The file having the changelog entry is expected to have a lines in the
+  # following format
+  #
+  #* #NNNNN <one line description of the issue/feature>
+  #
+  #* @name <Thank you note>  - optional
+  #
+  check_changelog_file:
+    name: Check for file with CHANGELOG entry
     runs-on: ubuntu-latest
     steps:
       - name: Checkout source
         uses: actions/checkout@v3
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
-      - name: Check CHANGELOG.md updated by pull request
+      - name: Check if the pull request adds file in ".unreleased" folder
         shell: bash --norc --noprofile {0}
         run: |
-          diff=$(git diff --name-status origin/main | grep "CHANGELOG.md")
-          cat << "EOF" | egrep -qsi '^disable-check:.*\<force-changelog-changed\>'
-          ${{ github.event.pull_request.body }}
-          EOF
-          if [[ $? -ne 0 ]]; then
-            if [[ $diff == '' ]]; then
-              echo
-              echo "CHANGELOG.md not updated by pull request"
+          folder=".unreleased"
```

Comment:
1. Have added the script to check the format of the non-empty lines.
2. Retained the check for changelog existence in the yaml, so that it would be quicker to analyse in case of failure.